strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/layout/npm/NpmArtifactControllerTest.java

```diff
@@ -125,4 +139,77 @@ public void testPackageCommonFlow(@NpmRepository(repositoryId = REPOSITORY_RELEA
                .header(HttpHeaders.CONTENT_LENGTH, equalTo(String.valueOf(Files.size(packagePath))));
     }
 
-}
+    @ExtendWith({ RepositoryManagementTestExecutionListener.class })
+    @ParameterizedTest
+    @ValueSource(strings = { "@angular/core:9.0.1-next.8",
+                             "react:0.0.0-experimental-5faf377df",
+                             "react:0.0.0-0c756fb-f7f79fd",
+                             "react:0.0.0-5faf377df",
+                             "prop-types:15.5.7-alpha.1",
+                             "commander:4.0.0-1",
+                             "aws-sdk:2.555.0",
+                             "rxjs:5.6.0-forward-compat.5",
+                             "@lifaon/observables:1.6.0" })
+    public void packageNameAcceptanceTest(String packageNameWithVersion,
+                                          @NpmRepository(repositoryId = REPOSITORY_RELEASES)
+                                          Repository repository,
+                                          TestInfo testInfo)
+            throws Exception
+    {
+        final String storageId = repository.getStorage().getId();
+        final String repositoryId = repository.getId();
+
+        //Splitting each string from paramterized test array to packageId and packageVersion
+        final String[] packageDetails = packageNameWithVersion.split(":");
+        final String packageId = packageDetails[0];
+        final String packageVersion = packageDetails[1];
+
+        //Manually generating each artifacts
+        generatorBasePath = Paths.get(propertiesBooter.getVaultDirectory(),
+                                      ".temp",
+                                      testInfo.getTestClass().get().getSimpleName(),
+                                      testInfo.getTestMethod().get().getName()).toAbsolutePath().normalize();
+
+        NpmArtifactGenerator artifactGenerator = new NpmArtifactGenerator(generatorBasePath);
+        Path artifact = artifactGenerator.generateArtifact(packageId, packageVersion);
+
+        NpmArtifactCoordinates coordinates = NpmArtifactCoordinates.of(packageId, packageVersion);
+
+        Path publishJsonPath = artifact.resolveSibling("publish.json");
+        byte[] publishJsonContent = Files.readAllBytes(publishJsonPath);
+
+        //Publish
+        String url = getContextBaseUrl() + "/storages/{storageId}/{repositoryId}/{artifactId}";
+        mockMvc.contentType(MediaType.APPLICATION_JSON_VALUE)
+                .body(publishJsonContent)
+                .when()
+                .put(url, storageId, repositoryId, coordinates.getId())
+                .peek()
+                .then()
+                .statusCode(HttpStatus.OK.value());
```

Comment:
Cosmetic: Could you please align these by the dot like this (here and everywhere below):
```
        mockMvc.contentType(MediaType.APPLICATION_JSON_VALUE)
               .body(publishJsonContent)
               .when()
               .put(url, storageId, repositoryId, coordinates.getId())
               .peek()
               .then()
               .statusCode(HttpStatus.OK.value());
```